org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xinclude/XIncludeErrorCode.java

```diff
@@ -0,0 +1,135 @@
+ */
+package org.eclipse.lemminx.extensions.xinclude;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.xerces.xni.XMLLocator;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.services.extensions.diagnostics.IXMLErrorCode;
+import org.eclipse.lemminx.utils.XMLPositionUtility;
+import org.eclipse.lsp4j.Range;
+
+/**
+ * XInclude error code.
+ *
+ * @see https://wiki.xmldation.com/Support/Validator
+ *
+ *      All error code types and messages can be found in the Xerces library
+ *      https://github.com/apache/xerces2-j/blob/trunk/src/org/apache/xerces/impl/msg/XIncludeMessages.properties
+ *
+ */
+public enum XIncludeErrorCode implements IXMLErrorCode {
+
+	NoFallback("NoFallback"), //
+	MultipleFallbacks("MultipleFallbacks"), //
+	FallbackParent("FallbackParent"), //
+	IncludeChild("IncludeChild"), //
+	FallbackChild("FallbackChild"), //
+	HrefMissing("HrefMissing"), //
+	RecursiveInclude("RecursiveInclude"), //
+	InvalidParseValue("InvalidParseValue"), //
+	XMLParseError("XMLParseError"), //
+	XMLResourceError("XMLResourceError"), //
+	TextResourceError("TextResourceError"), //
+	NonDuplicateNotation("NonDuplicateNotation"), //
+	NonDuplicateUnparsedEntity("NonDuplicateUnparsedEntity"), //
+	XpointerMissing("XpointerMissing"), //
+	AcceptMalformed("AcceptMalformed"), //
+	AcceptLanguageMalformed("AcceptLanguageMalformed"), //
+	RootElementRequired("RootElementRequired"), //
+	MultipleRootElements("MultipleRootElements"), //
+	ContentIllegalAtTopLevel("ContentIllegalAtTopLevel"), //
+	UnexpandedEntityReferenceIllegal("UnexpandedEntityReferenceIllegal"), //
+	HrefFragmentIdentifierIllegal("HrefFragmentIdentifierIllegal"), //
+	HrefSyntacticallyInvalid("HrefSyntacticallyInvalid"), //
+	XPointerStreamability("XPointerStreamability"), //
+	XPointerResolutionUnsuccessful("XPointerResolutionUnsuccessful");
+
+	private final String code;
+
+	private XIncludeErrorCode() {
+		this(null);
+	}
+
+	private XIncludeErrorCode(String code) {
+		this.code = code;
+	}
+
+	@Override
+	public String getCode() {
+		if (code == null) {
+			return name();
+		}
+		return code;
+	}
+
+	private final static Map<String, XIncludeErrorCode> codes;
+
+	static {
+		codes = new HashMap<>();
+		for (XIncludeErrorCode errorCode : values()) {
+			codes.put(errorCode.getCode(), errorCode);
+		}
+	}
+
+	public static XIncludeErrorCode get(String name) {
+		return codes.get(name);
+	}
+
+	/**
+	 * Create the LSP range from the SAX error.
+	 *
+	 * @param characterOffset
+	 * @param key
+	 * @param arguments
+	 * @param document
+	 * @return the LSP range from the SAX error.
+	 */
+	public static Range toLSPRange(XMLLocator location, XIncludeErrorCode code, Object[] arguments,
+			DOMDocument document) {
+		int offset = location.getCharacterOffset() - 1;
+		// adjust positions
+		switch (code) {
+			case NoFallback:
+			case XMLResourceError:
+				return XMLPositionUtility.selectAttributeValueAt("href", offset, document);
```

Comment:
`href` should be defined in constants. To be consistent with XSD, I suggest that you create an `XIIncludeUtils` and you define the `HREF_ATTR` like https://github.com/eclipse/lemminx/blob/df8eb888ee13b193aba6ff82322390967a0b1c66/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsd/utils/XSDUtils.java#L52

The main idea is to retrieve easily which features uses href attribute for XIInclude.

You will need this constants when we will implement documentLink for href attribute for instance.